### PR TITLE
OTM bid adapter: use top origin as default domain

### PR DESCRIPTION
**Pull request by regulyarniy**
## Type of change
- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
use top level origin as a default 'domain' argument for bid request

## Other information
Need @krecu review as the OTM module maintainer
Need @ChrisHuie review

### CODE REVIEW

**Review by krecu — APPROVED** [review]

**Comment by regulyarniy** [comment]
Need @ChrisHuie review

File: modules/otmBidAdapter.js [hunk 1]
```diff
@@ -132,15 +136,42 @@ function _getBidFloor(bid) {
...
   return 0;
 }
 
+/**
+ * Get top level origin
+ * @return {string}
+ */
+function getTopDomain() {
+  try {
+    return window.top.location.host
```
Comment (patmmccann) [inline modules/otmBidAdapter.js:156]: you can't do any of this; you have to use our methods.

**Review by patmmccann — CHANGES_REQUESTED** [review]
you have to use our methods

File: modules/otmBidAdapter.js (hunk 1 above)
Comment (regulyarniy) [reply modules/otmBidAdapter.js]: changed to `getWindowTop()`

**Comment by regulyarniy** [comment]
@patmmccann thx for review! Could you please also look why the build is failing? I think it's temporary problem with bs, nor the code problem.

File: modules/otmBidAdapter.js [hunk 2]
```diff
@@ -33,43 +46,37 @@ export const spec = {
...
 
     _each(validBidRequests, (bid) => {
-      let domain = getValue(bid.params, 'domain') || ''
-      let tid = getValue(bid.params, 'tid')
-      let cur = getValue(bid.params, 'currency') || DEF_CUR
-      let bidid = getBidIdParameter('bidId', bid)
-      let transactionid = getBidIdParameter('transactionId', bid)
-      let auctionid = getBidIdParameter('auctionId', bid)
-      let bidfloor = _getBidFloor(bid)
+      const domain = isStr(bid.params.domain) ? bid.params.domain : getTopDomain()
```
Comment (patmmccann) [inline modules/otmBidAdapter.js:53]: prebid already gets this for you with bidderRequest.refererInfo, which you grab above. Why are you adding this function?

Comment (patmmccann) [reply modules/otmBidAdapter.js:53]: in other words, https://github.com/prebid/Prebid.js/blob/master/src/refererDetection.js does this work for you, the rule to use our methods is to keep build sizes trim. If every adapter reproduced, build size would grow uncontrollably.

Comment (regulyarniy) [reply modules/otmBidAdapter.js]: now I see what u mean earlier, thx!
changed to use URL from `bidderRequest.refererInfo.referer`

**Review by patmmccann — APPROVED** [review]

**Pull request merged**